evm/test/propcorn/Propcorn.ts

```diff
@@ -2,9 +2,14 @@ import { SignerWithAddress } from "@nomicfoundation/hardhat-ethers/signers";
 import { loadFixture, time } from "@nomicfoundation/hardhat-network-helpers";
 import { expect } from "chai";
 import { parseEther } from "ethers";
-import { ethers } from "hardhat";
-
-import { Propcorn } from "../../types";
+import { ethers, upgrades } from "hardhat";
+
+import {
+  Propcorn,
+  PropcornNoMoreUpgrades__factory,
+  PropcornUpgraded,
+  PropcornUpgraded__factory,
```

Comment:
Are `PropcornUpgraded` and `PropcornNoMoreUpgrades` smart contracts? If so, where is the source?